### PR TITLE
Fix #1622: Circle-ci RDS snapshot replacement is buggy

### LINKED ISSUE #1622: CircleCI RDS snapshot replacement is buggy
### Issue

The CircleCI deployment fails at times because of the RDS snapshot replacement job: https://circleci.com/gh/hotosm/tasking-manager/2108?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

### Possible reasons
I think this particular failure was because two PRs were merged immediately in succession, and there was a race condition that failed because the two different circle CI instances tried to create an RDS snapshot with the same temp ID at the same time:
* https://circleci.com/gh/hotosm/tasking-manager/2106?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link succeeded
* https://circleci.com/gh/hotosm/tasking-manager/2108?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link failed because Circle CI 2106 succeeded and 2108 was trying to create the same snapshot as 2106, and started a little after 2106, and therefore failed.

This also meant the deploy did not go through for that particular commit, and could be the reason why the staging stack is wonky.

### Fix
I rebuilt the job: https://circleci.com/gh/hotosm/tasking-manager/2124, and it went through, so we could always do this, for an urgent workaround

### Solutions
For a longer-term solution, we could do the following:

Before the `Remove last snapshot and backup database` job in CircleCI, 
1. Check if the current GitSha is the `HEAD` GitSha of the branch.
2. If it is not don't do the next steps. If it is, check if the CloudFormation state of the corresponding TM stack is in `UPDATE_COMPLETE` state. If the stack is in any other state, wait for the stack to reach `UPDATE_COMPLETE` before continuing those steps.

cc/ @dakotabenjamin

### PR DESCRIPTION
**Pull request by eternaltyro**
Introduce a random string nonce for the temporary snapshot name to
avoid collision.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -166,18 +166,27 @@ jobs:
...
+          echo "export NONCE=$NONCE" >> $BASH_ENV
           # Copy old snapshot to temporary
           if [ -z "$DESCRIBE_SNAPSHOT" ]
           then
               echo "Snapshot does not exist, creating one now."
           else
-              aws rds copy-db-snapshot --source-db-snapshot tm3-<< parameters.stack_name >>-$RDS_ID-latest --target-db-snapshot tm3-<< parameters.stack_name >>-$RDS_ID-temp
-              aws rds delete-db-snapshot --db-snapshot-identifier tm3-<< parameters.stack_name >>-$RDS_ID-latest
+              aws rds copy-db-snapshot \
+                      --source-db-snapshot tm3-<< parameters.stack_name >>-$RDS_ID-latest \
```
Comment (dakotabenjamin) [inline .circleci/config.yml]: Should `$RDS_ID` be in `{}` here for consistency?

Comment (eternaltyro) [reply .circleci/config.yml]: All bash variables throughout should ideally be within `{}` for consistency. I'll change it.

Comment (eternaltyro) [reply .circleci/config.yml]: Done.

**Comment by xamanu** [comment]
Shall we make a hotfix release out of this? Or wait for the next release, which might take some time?

**Comment by dakotabenjamin** [comment]
It does not need to be a hot fix

**Pull request merged**